programs/clearing_house/src/math/amm.rs

```diff
@@ -1842,80 +1849,63 @@ mod test {
-        let (t_price, _t_qar, _t_bar) = calculate_terminal_price_and_reserves(&market).unwrap();
+        let (t_price, _t_qar, _t_bar) = calculate_terminal_price_and_reserves(&market.amm).unwrap();
 
         // new terminal reserves are balanced, terminal price = peg)
         // assert_eq!(t_qar, 999900009999000);
         // assert_eq!(t_bar, 1000100000000000);
-        assert_eq!(t_price, 499001498002497); //
+        assert_eq!(t_price, 499011369495392); //
                                               // assert_eq!(update_k_up.sqrt_k, 101 * AMM_RESERVE_PRECISION);
 
         let cost = adjust_k_cost(&mut market, &update_k_up).unwrap();
         assert_eq!(
             market.amm.net_base_asset_amount,
             (AMM_RESERVE_PRECISION / 10) as i128
         );
-        assert_eq!(cost, 50_404); //0.05
+        assert_eq!(cost, 49406); //0.05
 
         // lp whale adds
         let lp_whale_amount = 1000 * AMM_RESERVE_PRECISION;
-
-        position.lp_shares += lp_whale_amount;
-        let update_k_lp_whale = get_update_k_result(
-            &market,
-            bn::U192::from(market.amm.sqrt_k + lp_whale_amount),
-            false,
-        )
-        .unwrap();
-        update_k(&mut market, &update_k_lp_whale).unwrap();
-        market.amm.user_lp_shares += lp_whale_amount;
+        mint_lp_shares(&mut position, &mut market, lp_whale_amount, 0).unwrap();
 
         // ensure same cost
         let update_k_up =
-            get_update_k_result(&market, bn::U192::from(1101 * AMM_RESERVE_PRECISION), false)
+            get_update_k_result(&market, bn::U192::from(1102 * AMM_RESERVE_PRECISION), false)
                 .unwrap();
         let cost = adjust_k_cost(&mut market, &update_k_up).unwrap();
         assert_eq!(
             market.amm.net_base_asset_amount,
             (AMM_RESERVE_PRECISION / 10) as i128
         );
-        assert_eq!(cost, 50_404); //0.05
+        assert_eq!(cost, 49406); //0.05
 
         let update_k_down =
             get_update_k_result(&market, bn::U192::from(1001 * AMM_RESERVE_PRECISION), false)
                 .unwrap();
         let cost = adjust_k_cost(&mut market, &update_k_down).unwrap();
-        assert_eq!(cost, -4994954591); //amm rug
+        assert_eq!(cost, -4995004995); //amm rug
 
         // lp whale removes
-        position.lp_shares -= lp_whale_amount;
-        let update_k_lp_whale = get_update_k_result(
-            &market,
-            bn::U192::from(market.amm.sqrt_k - lp_whale_amount),
-            false,
-        )
-        .unwrap();
-        update_k(&mut market, &update_k_lp_whale).unwrap();
-        market.amm.user_lp_shares -= lp_whale_amount;
+        burn_lp_shares(&mut position, &mut market, lp_whale_amount, 0).unwrap();
 
         // ensure same cost
         let update_k_up =
-            get_update_k_result(&market, bn::U192::from(101 * AMM_RESERVE_PRECISION), false)
+            get_update_k_result(&market, bn::U192::from(102 * AMM_RESERVE_PRECISION), false)
```

Comment:
why did we change this?

Reply:
sqrt_k starts at 100 - you add lp of 1 share - sqrt_k is 101 - to compute the update k cost we do it with 102 or the cost = 0